### PR TITLE
Fix game crashes

### PR DESCRIPTION
**Pull request by V-rtualized**
- Game crashes when server times out
- Playing singleplayer crashes when using the Nebula deck
- Game crashes for the one who leaves a lobby

### CODE REVIEW

File: Multiplayer/Lobby.lua [hunk 1]
```diff
@@ -73,5 +73,12 @@ function G.MULTIPLAYER.update_player_usernames()
 	end
 end
 
+
+local wipe_off_ref = G.FUNCS.wipe_off
+G.FUNCS.wipe_off = function()
+	if not G.screenwipe then return end
+	wipe_off_ref()
+end
```
Comment (V-rtualized) [inline Multiplayer/Lobby.lua:81]: This probably prevents the crash I was getting (`G.screenwipe` is nil in `wipe_off`) but I am unable to reproduce the crash consistently so I am not confident

**Comment by V-rtualized** [comment]
Also starting a game using the Nebula deck does not crash my game, is this fixed or is it more complicated/inconsistent? @TGMM

**Comment by TGMM** [comment]
> Also starting a game using the Nebula deck does not crash my game, is this fixed or is it more complicated/inconsistent? @TGMM

I think we might've unintentionally fixed this when we added singleplayer checks for some of our functions

File: Multiplayer/Lobby.lua (hunk 1 above)
Comment (TGMM) [reply Multiplayer/Lobby.lua:81]: I'm still getting the crash on timeout. These are the repro steps:
- Connect to the server
- Create or join a lobby
- Close the server
It'll try to return you back to the main menu and crash on the loading screen

Comment (V-rtualized) [reply Multiplayer/Lobby.lua:81]: Ok, this is because action_disconnect is setting `G.LOBBY.code = nil` and `G.FUNCS.go_to_menu()`. In the gameloop we are checking for changes in `G.LOBBY.code` and if it becomes `nil` then it calls `G.FUNCS.go_to_menu()`. The crash is a result of both of those attempting to happen at the same time. (Because the source code excessively uses async functions)

TL;DR if you want to leave the lobby just set the code to nil, no need to do anything else (I think this was caused by my code though)

Comment (TGMM) [reply Multiplayer/Lobby.lua:81]: Seems fixed now, good job! I'm wondering if we could do something about the race condition for starting a game. A similar thing occurs where if you create a lobby and have your someone join you, when you press Start while the one who joined the lobby is in the loading screen, it crashes the game for them.

Not saying it should be a part of this PR, but we could investigate further.

**Review by TGMM — APPROVED** [review]

File: Multiplayer/Lobby.lua (hunk 1 above)
Comment (V-rtualized) [reply Multiplayer/Lobby.lua:81]: Yeah, I tried to implement a mutex for `wipe_on` and `wipe_off` but it didn't work correctly because those functions just call async events, but with a little more attention we could definitely do that in the future
